src/Symfony/Component/Security/Core/User/CurrentUserProvider.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Symfony\Component\Security\Core\User;
+
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+/**
+ * Current User Provider.
+ *
+ * This provider gives you the current logged in user.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class CurrentUserProvider
```

Comment:
I'm not sure if it's a good idea to add a class just to wrap the TokenStorage.

In my opinion, the token/user shouldn't be called in your application controller, they should only be used to identify the current user and can be used to fetch the information of the current user.

Next to all this, there is already a PR open in the sensio framework-extra-bundle repo regarding a ParamConverter to inject the current user. This is a cleaner solution in my opinion: sensiolabs/SensioFrameworkExtraBundle/pull/327